app/app/private-cloud/requests/(request)/[id]/comments/page.tsx

```diff
@@ -0,0 +1,125 @@
+'use client';
+
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import React, { useEffect } from 'react';
+import { z } from 'zod';
+import CommentBubble from '@/components/comments/CommentBubble';
+import CommentForm from '@/components/comments/CommentForm';
+import createClientPage from '@/core/client-page';
+import { PrivateCloudRequestGetPayload } from '@/queries/private-cloud-requests';
+import { getAllPrivateCloudComments } from '@/services/backend/private-cloud/products';
+import { getPrivateCloudRequest } from '@/services/backend/private-cloud/requests';
+import { privateProductState, usePrivateProductState } from '@/states/global';
+
+interface User {
+  firstName: string;
+  lastName: string;
+  email: string;
+}
+
+interface Comment {
+  id: string;
+  createdAt: Date;
+  updatedAt: Date;
+  text: string;
+  userId: string;
+  user: User;
+}
+
+const pathParamSchema = z.object({
+  id: z.string(),
+});
+
+const privateCloudRequestComments = createClientPage({
+  roles: ['user'],
+  validations: { pathParams: pathParamSchema },
+});
+
+export default privateCloudRequestComments(({ pathParams, session }) => {
+  const queryClient = useQueryClient();
+  const [privateState, privateSnap] = usePrivateProductState();
+  const { id: requestId } = pathParams;
+  const userId = session?.userId;
+
+  // Fetch the request details to ensure licencePlate is set
+  const {
+    data: request,
+    isLoading: requestLoading,
+    isError: requestError,
+    error: requestErrorDetails,
+  } = useQuery<PrivateCloudRequestGetPayload>({
+    queryKey: ['request', requestId],
+    queryFn: () => getPrivateCloudRequest(requestId),
```

Comment:
request should be available from the higher layout level so that you can get the data from `privateProductState`

Reply:
The issue I was having was that the comments would not load in from a `browser refresh`, but when I switched from any of the request tabs to the `ADMIN COMMENTS` tab e.g I am on the  `ADMIN DECISION` tab I then click on the `ADMIN COMMENTS` tab then the comments load in no problem. This was the solution to the problem, as the second commit of this PR.

Reply:
I will take a look.